File: drl_sample_project/src/do_not_touch/result_structures.rs
use std::collections::HashMap;

/// Pi(s,a)
#[derive(Debug, Clone)]
pub struct Policy(HashMap<usize, HashMap<usize, f32>>);

/// V(s)
#[derive(Debug, Clone)]
pub struct ValueFunction(HashMap<usize, f32>);

/// Q(s,a)
#[derive(Debug, Clone)]
pub struct ActionValueFunction(HashMap<usize, HashMap<usize, f32>>);

/// Pi(s,a) and V(s)
#[derive(Debug, Clone)]
pub struct PolicyAndValueFunction {
    pi: Policy,
    v: ValueFunction,
}

/// Pi(s,a) and Q(s,a)
#[derive(Debug, Clone)]
pub struct PolicyAndActionValueFunction {
    pi: Policy,
    q: ActionValueFunction,
}

File: drl_sample_project_python/do_not_touch/result_structures.py
# Pi(s,a)
from dataclasses import dataclass
from typing import Dict

Policy = Dict[int, Dict[int, float]]

# V(s)
ValueFunction = Dict[int, float]

# Q(s,a)
ActionValueFunction = Dict[int, Dict[int, float]]


# Pi(s,a) and V(s)
@dataclass
class PolicyAndValueFunction:
    pi: Policy
    v: ValueFunction


# Pi(s,a) and Q(s,a)
@dataclass
class PolicyAndActionValueFunction:
    pi: Policy
    q: ActionValueFunction
